src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -161,15 +161,18 @@ def forward(self, *inputs):
             *inputs: Input to the module.
 
         Returns:
-            torch.Tensor: Module output.
+            torch.distributions.independent.Independent: Independent
```

Comment:
Is this actually returning `torch.distributions.Independent`? Shouldn't it be returning `torch.distributions.Normal`?

Reply:
I believe the function return torch.distributions.independent.Independent.

https://github.com/rlworkgroup/garage/blob/e7d6db3c5caff27e53c46e7cbb105c55c41f3953/src/garage/torch/modules/gaussian_mlp_module.py#L184

Reply:
@avnishn can you explain why this is happening?

Reply:
So if we return a `torch.distributions.Normal` then the shape of the `log_likelihood` calculation would be incorrect as far as I know. Wrapping the underlying gaussian with the Independent wrapper allows us to change the shape of `log_likelihood` calculation to the same shape as if we were using a `torch.distributions.MultivariateNormal`.

[https://pytorch.org/docs/stable/distributions.html#independent](Documentation of the torch Independent wrapper explains what I said above along with a code example for further clarity)

Reply:
can you please be more specific instead of saying "incorrect" and "as if we were using" -- what input-output shapes are you getting and what do you want them to be instead?

Reply:
![image](https://user-images.githubusercontent.com/38871737/77237561-9f435080-6b86-11ea-8a3f-de6712f9dfb8.png)

Reply:
If I construct a multivariate normal distribution, then when I compute the log probability of a sample on the distribution, then a single log probability is associated with this sample. 

If I construct a normal distribution however with the same shape mean and standard deviation, then when I compute the log probability of a random sample on that distribution, then torch treats the sample as 3 samples coming from 3 distributions, as opposed to a shape (3,) sample coming from a single distribution, and so what is returned is 3 log probabilities. This is incorrect.

If I wrap the normal distribution with the Independent wrapper, then I fix the above problem, and receive the memory/speed benefits of using a normal distribution, while getting the functionality of diagonal multivariate gaussian normal distribution.

Reply:
so, basically, `Normal` doesn't support batches of multivariate data?

Reply:
okay. this is a dumb API choice on Torch's part.

please document it like a collection, showing us what Independent is wrapping, e.g. `Indepedent[Normal]`